### PR TITLE
[FIX] web: allow user to scroll horizontally

### PR DESCRIPTION
**Pull request by yosa-odoo**
Steps to reproduce:
- create a contact with a very long name
- create a quotation
- in customer, type the first letter of the very long customer's name
- try to use the horizontal scroll bar

Issue:
close window on horizontal scroll

Solution:
Discriminate an horizontal scroll from a vertical scroll

opw-2777443
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Iucapad** [comment]
You should add a test to check that you can effectively scroll horizontally without closing  the list of items. Just a note but we are currently remaking fields in Owl. 
Adding a test would assure that the behavior is maintained while we reimplement the field (I just checked and the Owl field works fine but more tests is always better :) )

**Comment by Iucapad** [comment]
Also, I think it would be better if the fix don't just allow horizontal scrolling but scrolling inside the dropdown if the content is larger/higher than the container. 
And in the tests: 
- keep the test as it is by listening to scroll event globally to close the dropdown. 
- add a test to check that you can scroll inside the container if needed.

**Comment by yosa-odoo** [comment]
Thanks for the review, I'm on it!

**Comment by Iucapad** [comment]
@yosa-odoo So I added a new utility to trigger a scroll by giving a target and the x and y attributes related to the scroll. 
I also added a test to more specifically test scrolling inside the element when it can be scrolled and when it cannot.

@aab-odoo can you review the changes?

**Comment by yosa-odoo** [comment]
@lucapad good! I was still getting an error for the horizontal scroll for the specific and still investigating why. Thanks!

File: addons/web/static/src/legacy/js/fields/relational_fields.js [hunk 1]
```diff
@@ -316,7 +316,7 @@ var FieldMany2One = AbstractField.extend({
             },
             open: function (event) {
                 self._onScroll = function (ev) {
-                    if (ev.target !== self.$input.get(0) && self.$input.hasClass('ui-autocomplete-input')) {
+                    if (ev.target !== self.$input.get(0) && self.$input.hasClass('ui-autocomplete-input') && ev.target.scrollTop != 0) {
```
[the remark this answers is not in the document]
Comment (aab-odoo) [reply addons/web/static/src/legacy/js/fields/relational_fields.js:319]: And closing the dropdown only when the target isn't the dropdown itself

File: addons/web/static/tests/legacy/fields/relational_fields/field_many2one_tests.js [hunk 2]
```diff
@@ -3315,12 +3315,54 @@ QUnit.module('fields', {}, function () {
...
 
             var $input = form.$('.o_field_many2one input');
+            var dropdown = document.querySelector(".dropdown-menu.ui-front");
 
             await testUtils.dom.click($input);
             assert.isVisible($input.autocomplete('widget'), "dropdown should be opened");
 
-            form.el.dispatchEvent(new Event('scroll'));
-            assert.isNotVisible($input.autocomplete('widget'), "dropdown should be closed");
+            await triggerScroll(dropdown, { top: 50 }, false);
```
Comment (aab-odoo) [inline addons/web/static/tests/legacy/fields/relational_fields/field_many2one_tests.js]:
```suggestion
```

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**